### DISCUSSION TITLE
Migrate the deprecated OnBackPressed function in .NET MAUI Android application

### DISCUSSION THREAD

**Discussion by giscus** [VladislavAntonyuk/vladislavantonyuk.github.io#96] [Announcements]
# Migrate the deprecated OnBackPressed function in .NET MAUI Android application

This article is devoted to the new API added in Android 13, that adds support to predictive back gestures.

https://vladislavantonyuk.github.io/articles/Migrate-the-deprecated-OnBackPressed-function-in-.NET-MAUI-Android-application/

**Comment by vitalbit** [comment]
Probably one important note to add here. 

If you have page navigation and need to track back button pressed on the navigated page, you need to set the following in AndroidManifest.xml:
<application ... **android:enableOnBackInvokedCallback="true"**>

Works fine with the latest MAUI update (8.0.10).

**Reply by VladislavAntonyuk** [reply]
I don't think it is required, because you add callback OnCreate in Activity: `OnBackPressedDispatcher.AddCallback(this, new BackPress(this));`

**Reply by vitalbit** [reply]
Haven't worked for me when I used AppShell navigation and wanted to track back button click from the page navigated to by routing. Only **android:enableOnBackInvokedCallback="true"** helped me.
